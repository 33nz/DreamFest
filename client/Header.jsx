const React = require('react')

function Header (props) {
  const {title, link} = props.content
  return (
    <div className='header'>
      <h1><a href={link}>{title}</a></h1>
    </div>
  )
}

module.exports = Header

