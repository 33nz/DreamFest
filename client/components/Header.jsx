import React from 'react'
import { Link } from 'react-router-dom'

import Nav from './Nav/Nav'
import WaitIndicator from '../components/WaitIndicator/WaitIndicator'

export default function Header () {
  return (
    <header className="header-container">
      <div className='container'>
        <Link className="logo-wrapper" to="/">
          <img src='/images/gardenzLogo.svg' alt="gardenzlogo" className='logo-image' />
        </Link>
        <WaitIndicator />
        <Nav />
      </div>
    </header>
  )
}
