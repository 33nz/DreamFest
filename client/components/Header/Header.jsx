import React, { useState } from 'react'

import Logo from '../Logo/Logo'
import Nav from '../Nav/Nav'

const Header = () => {
  const [navOpen, setNavOpen] = useState(false)

  const toggleNav = () => {
    setNavOpen(!navOpen)
  }

  return (
    <header className='app-header'>
      <Logo />
      <button className='app-header__nav-toggle' onClick={toggleNav}>
        <span className={`fa fa-${navOpen ? 'times' : 'bars'}`}></span>
      </button>
      <nav className={`nav${navOpen ? ' nav--open' : ''}`} onClick={toggleNav}>
        <Nav />
      </nav>
    </header>
  )
}

export default Header
