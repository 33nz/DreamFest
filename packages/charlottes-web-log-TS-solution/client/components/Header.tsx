import type * as type from '../data/header'

const Header = ({ title }: type.Header) => {
  return <h2>{title}</h2>
}

export default Header