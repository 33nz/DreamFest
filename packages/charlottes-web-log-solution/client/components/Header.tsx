import header from '../data/header'

export default function Header() {
  return (
    <div className="header">
      <h1>
        <a href={header.link}>{header.title}</a>
      </h1>
    </div>
  )
}
