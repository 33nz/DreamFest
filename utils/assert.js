var test = require('tape')
var tapSpec = require('tap-spec')

test.createStream()
  .pipe(tapSpec())
  .pipe(process.stdout)

function assert(expect, actual, msg) {
  if (typeof actual === 'undefined') {
    console.log('parameter "actual" is undefined!')
  }
  test(msg || "please add a message",  function (t) {
    t.equal(expect, actual)
    t.end()
  })
}

module.exports = assert
