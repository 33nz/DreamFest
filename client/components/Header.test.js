import React from 'react'
import { screen } from '@testing-library/react'
import { renderWithRouter } from '../testing/utils'
import '@testing-library/jest-dom'

import Header from './Header'

test('renders header correctly', async () => {
  const { asFragment } = renderWithRouter(<Header />)

  const nav = await screen.getByRole('navigation')
  const header = await screen.getByRole('heading')

  expect(nav.children).toHaveLength(3)
  expect(header).toHaveTextContent('Sweet As Organics')
  expect(asFragment()).toMatchSnapshot()
})
