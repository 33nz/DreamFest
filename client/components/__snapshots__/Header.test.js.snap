// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header correctly 1`] = `
<DocumentFragment>
  <nav
    class="nav"
  >
    <a
      href="/"
    >
      Shop
    </a>
    <a
      href="/cart"
    >
      Cart
    </a>
    <a
      href="/orders"
    >
      My Orders
    </a>
  </nav>
  <h1>
    <span
      aria-hidden="true"
      class="fa fa-leaf"
    />
     Sweet As Organics 
    <span
      aria-hidden="true"
      class="fa fa-leaf"
    />
  </h1>
</DocumentFragment>
`;
